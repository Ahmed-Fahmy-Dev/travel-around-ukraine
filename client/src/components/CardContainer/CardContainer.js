import React from 'react';
import {NavLink} from "react-router-dom";
import {Swiper, SwiperSlide} from "swiper/react";
import { Navigation} from 'swiper'
import CardItem from '../CardItem/CardItem';
import styles from './CardContainer.module.scss';

import "swiper/css";
import "swiper/css/pagination";
import "swiper/css/navigation";



const cards = [
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
    <CardItem/>,
]

const CardContainer = () =>
    <div className={styles.cardContainer}>
        <div className={styles.headerContainer}>
            <h2 className={styles.header}>Popular tours</h2>
            <NavLink className={styles.allTours} to="/catalogue">View All</NavLink>
        </div>
                <Swiper
                    slidesPerView={4}
                    spaceBetween={30}
                    slidesPerGroup={1}
                    loop={true}
                    loopFillGroupWithBlank={true}
                    pagination={{
                        clickable: true,
                    }}
                    navigation={true}
                    modules={[Navigation]}
                    className="mySwiper">
                    {cards.map(card => (
                        <SwiperSlide
                        >{card}</SwiperSlide>
                    ))}
                </Swiper>
    </div>

export default CardContainer;

