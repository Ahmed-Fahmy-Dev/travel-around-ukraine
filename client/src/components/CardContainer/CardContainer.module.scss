.cardContainer {
  max-width: 1150px;
  margin: 0 auto;
  padding: 0 20px;
}

.headerContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;

  .header {
    font-size: 24px;
    color: #0499DD;
    margin-bottom: 0;
  }

  .allTours {
    font-size: 12px;
    color: #282828;
  }
}

.swiperContainer {
  position: relative;
  margin: 70px auto 0 auto;

  .btnPrev {
    background-color: white;
    height: 60px;
    width: 60px;
    border-radius: 50px;
    position: absolute;
    cursor: pointer;
    z-index: 2;
    top: 50%;
    margin-top: -30px;
    margin-left: -30px;

    &::after {
      content: url("../../../src/svg/hotDeals/btn-prev.svg");
      position: absolute;
      width: 10px;
      top: 50%;
      left: 50%;
      margin-left: -2px;
      margin-top: -9px;
      transform: scaleX(-1);
    }
  }

  .btnNext {
    background-color: white;
    height: 60px;
    width: 60px;
    border-radius: 50px;
    position: absolute;
    cursor: pointer;
    z-index: 2;
    top: 50%;
    right: -30px;
    margin-top: -30px;
    margin-left: 100%;

    &::after {
      content: url("../../../src/svg/hotDeals/btn-prev.svg");
      position: absolute;
      width: 10px;
      top: 50%;
      left: 50%;
      margin-left: -6px;
      margin-top: -9px;
    }
  }
}

@media only screen and (min-width: 300px) {
  .swiperContainer {
    width: 265px;
  }
}

@media only screen and (min-width: 600px) {
  .swiperContainer {
    width: 560px;
  }
}

@media only screen and (min-width: 900px) {
  .swiperContainer {
    width: 860px;
  }
}

@media only screen and (min-width: 1200px) {
  .swiperContainer {
    width: 1150px;
  }
}

