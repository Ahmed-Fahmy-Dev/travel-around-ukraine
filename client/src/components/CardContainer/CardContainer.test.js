import {render} from '@testing-library/react';
import CardContainer from './CardContainer'
import {MemoryRouter, Router} from 'react-router-dom'
import React from "react";

const RouterWrapper = ({children}) => <MemoryRouter>{children}</MemoryRouter>;

describe('Card Container Snapshot test', () => {
    test('should CardContainer match snapshot', () => {
        const {asFragment} = render(<CardContainer/>, {wrapper: RouterWrapper});
        expect(asFragment()).toMatchSnapshot();
    });
});