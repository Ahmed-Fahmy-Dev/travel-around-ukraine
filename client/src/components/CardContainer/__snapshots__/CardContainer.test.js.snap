// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Container Snapshot test should CardContainer match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="css-4b3544"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-cf795h-MuiTypography-root"
      >
        Popular tours
      </h2>
      <a
        href="/catalogue"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          View All
        </p>
      </a>
    </div>
    <div
      class="css-1xiibfn"
    >
      <div
        class="css-1on216r"
        id="previous"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </div>
      <div
        class="css-1tjowg6"
        id="next"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </div>
      <div
        class="swiper swiper-initialized swiper-horizontal swiper-pointer-events mySwiper"
      >
        <div
          class="swiper-wrapper"
          style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide swiper-slide-duplicate"
            data-swiper-slide-index="7"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="0"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="1"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="2"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="3"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="4"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="5"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="6"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="7"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide swiper-slide-duplicate"
            data-swiper-slide-index="0"
          >
            <div
              class="css-1iryrks"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ymd0vv-MuiTypography-root"
              >
                $ 200.00
              </p>
              <h3
                class="MuiTypography-root MuiTypography-h3 css-1w7kxm5-MuiTypography-root"
              >
                Kyiv city walk
              </h3>
              <div
                class="css-h56svl"
              >
                <img
                  alt="location"
                  class="MuiBox-root css-dyqcbf"
                  src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1djbt89-MuiTypography-root"
                >
                  Kyiv region
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
                  data-testid="ArrowForwardIosIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="swiper-pagination"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
